src/mqtt/mod.rs

```diff
@@ -218,6 +225,101 @@ async fn listen_on_camera(cam_config: Arc<CameraConfig>, mqtt_config: &MqttConfi
     Ok(())
 }
 
+/// Enables MQTT discovery for a camera. See docs at https://www.home-assistant.io/integrations/mqtt/#mqtt-discovery
+async fn enable_discovery(
+    discovery_config: &MqttDiscoveryConfig,
+    mqtt_sender: &MqttSender,
+    cam_config: &Arc<CameraConfig>,
+) -> Result<()> {
+    debug!("Enabling MQTT discovery for {}", cam_config.name);
+
+    let mut connections = array![];
+    if let Some(addr) = &cam_config.camera_addr {
+        connections
+            .push(array!["camera_addr", addr.clone()])
+            .expect("Failed to add camera_addr to connections");
+    }
+    if let Some(uid) = &cam_config.camera_uid {
+        connections
+            .push(array!["camera_uid", uid.clone()])
+            .expect("Failed to add camera_uid to connections");
+    }
+
+    let friendly_name = cam_config.name.replace("_", " ").to_title_case();
+    let device = object! {
+        connections: connections,
+        name: friendly_name.clone(),
```

Comment:
rust advice: do i need to `.clone()` this string every time? or is there some better option?

Reply:
You should proabably be using references instead or `.as_str()` but will need time to understand ownership.

For example later on here you are cloning the device multiple times. You only have one read only copy so a reference to it would be better then a clone.

Reply:
OK, I took a shot at reducing copying here. 

`device` and `availability` were the most complicated because they are `JsonValue`. Not sure if this approach actually makes sense or not vs defining an intermediate `struct` instead then converting it every time? Maybe that's better...

Reply:
Reducing copying with `device` and `availability` didn't really work as I hoped given the JSON values. Personally I think this is fine to keep as-is because it's only run once on initialization. Let me know if you disagree.